### ISSUE TITLE
loop condition

### ISSUE THREAD

**Issue by engFelipeMonteiro** [rangav/thunder-client-support#1199]
**Please describe feature/problem details and solution you'd like.**
I have a test case that need to wait until some result change, in Postman I check the result and set next request for the same request until iit has changed.

**Describe alternatives you've considered**
I  think it have some approach:
1. In 'Pre Run', a new condition 'while variable is Empty', that repeat a request while a variable not set.
2. A nextRequest function on the 'tc' object to determine the next request in the workflow of the testCase

I have tried to make from a test scripting, but I cant surpass the SSL error in the axios library.

**Comment by rangav** [comment]
Hi @engFelipeMonteiro thanks for the feedback, Will review it.

Regarding axios ssl see below example.

```js
const agent = new https.Agent({  
  rejectUnauthorized: false
});
axios.get('https://something.com/foo', { httpsAgent: agent });
```

**Comment by rangav** [comment]
Hi @engFelipeMonteiro I think you can use Run Collection to loop through the same request until a value changes

1. Set Iteration to 100
2. Select `Stop On Fail`
3. From the Tests tab create a test that will fail when values change
4. Run the Collection, when the test fails, the execution of the collection will stop

![image](https://github.com/rangav/thunder-client-support/assets/8637550/aebf59e0-aecb-4c25-958a-0e7e98a0299f)

**Comment by engFelipeMonteiro** [comment]
Thanks for the fast reply.


> Hi @engFelipeMonteiro I think you can use Run Collection to loop through the same request until a value changes
> 
> 1. Set Iterations to 100
> 2. Select `Stop On Fail`
> 3. Open the request and from the Tests tab create a test that will fail when values change
> 4. Run the Collection, when the test fails, the execution of the collection will stop
> 
> ![image](https://user-images.githubusercontent.com/8637550/246369983-aebf59e0-aecb-4c25-958a-0e7e98a0299f.png)

Imagine that I have a test case that have 3 requests  (POST to create a process and 2 GET(first I check if the process is completed and the second I in fact get value needed) ), the POST take some time to complete so I need to run the first GET until it's return value change or until the script change a env value, than I run the last GET

**Comment by engFelipeMonteiro** [comment]
I think that it would resolve my case.
what is the prevision for this new feature?
I am looking forward of this

**Comment by rangav** [comment]
This feature is now implemented and published to the marketplace, please update to `v2.9.0`

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.9.0

Please let me know your feedback.
